### PR TITLE
Feature fav btn ajax

### LINKED ISSUE #50: Reconsider Favoriting Stops Redirecting to Favorites
Favoriting a stop redirects you to the favorites page, which takes you away from any page you might have been on. Favoriting should probably fire an AJAX request instead or at worst refresh the current page.

Original report from UX review last week:
> When clicking on the Favorite button, it automatically takes me to the Favorites page (which I don't necessarily want).

### PR DESCRIPTION
**Pull request by Thrilldozer**
I would like to resolve #50. 

This adds the jquery to send an ajax request that updates the visuals to show when its favorited.  

added "db/test.sqlite3" to the database.yml because I had to create a stop in the tests I wrote for the favorites controller.

### CODE REVIEW

**Comment by vkoves** [comment]
Thank you for this @Thrilldozer! I'm excited to take a deeper look into this, and I appreciate your tests around the favorites feature.

The one suggestion I would have is to use `.js.erb` for handling the JS callback. Using `ajax:success` works well when only handling one type of ajax request, but any ajax requests made would call the handler on `.stop-partial a`, which would probably cause an error.

To future proof this code and follow Ruby on Rails best practice, we should instead have the server return a JS file (which will be in code as a `.js.erb` file) that will execute when a request succeeds for favoriting and will take care of the UI behavior we expect. The Ruby On Rails guides have a good example of this in their "Working with JavaScript in Rails" guide, found [here](http://guides.rubyonrails.org/working_with_javascript_in_rails.html#a-simple-example). Could you try and implement your callback so it is in a `.js.erb` file as they suggest?

Thanks again!

**Comment by vkoves** [comment]
@Thrilldozer - is this ready for re-review?

**Comment by Thrilldozer** [comment]
it will be today.  I had some life issues come up this week but its ready
to go I just have a few things to clean up

**Comment by vkoves** [comment]
@Thrilldozer - wanted to check in on this again. Something seems to have happened with the commits, which is a bit puzzling, so I wanted to see if you knew anything about that. Otherwise, if this isn't ready yet, just wanted to say that you can tag me in a comment when it is and I'll take a look. Thanks!

**Comment by Thrilldozer** [comment]
I have to be honest this is my first time working on a project this big so I may be a bit confused about submitting my work.  I thought I had submitted it for review with the Update branch button lol.  I think its ready for review but if Im missing a step to get it put up just let me know.

File: app/models/stop.rb [hunk 1]
```diff
@@ -16,4 +16,9 @@ class Stop < ApplicationRecord
   def original
     twin_stop || self
   end
+
+  def html_id
```
Comment (vkoves) [inline app/models/stop.rb]: Can you add a comment explaining the intent of this function? Also, should we have tests around this function?

File: app/views/favorites/index.html.erb [hunk 2]
```diff
@@ -2,8 +2,8 @@
 	<h1>Favorite Stops</h1>
 	<% unless current_user.favorites.empty? %>
 		<% current_user.favorites.each do |stop| %>
-			<%= render "stops/show_with_issues", stop: stop %>
-		<% end %>
+		  <%= render "stops/show_with_issues", stop: stop %>
```
Comment (vkoves) [inline app/views/favorites/index.html.erb]: It looks like some indenting got changed here that shouldn't have been. Could you remedy it?

**Comment by Thrilldozer** [comment]
I lost track but I started back up on it. I want to have it done by Monday
but with the holiday weekend it may take a bit longer. Sorry for the wait
lol

**Comment by vkoves** [comment]
@Thrilldozer - no worries, thanks for fixing it up and for working on this!

**Comment by vkoves** [comment]
This pull request has now been stale for a long time, but the work here is valuable, so I'll see if I can fix it up. I'll start by trying to fix the history with `git cherry-pick` and then see about fixing the issue I found.

**Comment by vkoves** [comment]
Closing this as it is out of date and this was solved via other means.

**Pull request closed without merging**